model/src/main/java/com/flipkart/flux/domain/Context.java

```diff
@@ -32,7 +33,7 @@
     /** Identifier for the Context*/
     protected String contextId;
     /** A dependency graph created across States - holds information on possible next state transitions for a State, keyed by the Event FQN*/
-    protected Map<String,List<State<T>>> stateToEventDependencyGraph;
+    protected Map<String, Map<String, State<T>>> stateToEventDependencyGraph;
```

Comment:
No, its a reverse dependency graph. 
It means that a given event (identified by string for now) has the following list of states waiting on it.
Once we get the event, we can iterate over the list and figure if any of the states is unblocked.
We can enter all unblocked states at once

Also, a task will raise only event. The return value of the method that is executed as per the task execution is wrapped in an event and sent back to the runtime